src/scatter_buildblock/ScatterEstimation.cxx

```diff
@@ -0,0 +1,1486 @@
+                                          this->input_projdata_sptr->get_proj_data_info_sptr() ,
+                                          out_filename,
+                                          std::ios::in | std::ios::out | std::ios::trunc));
+        add_projdata_sptr->fill(*back_projdata_sptr);
+        this->multiplicative_binnorm_sptr->apply(*this->add_projdata_sptr, start_time, end_time);
+
+        iterative_object->get_objective_function_sptr()->set_additive_proj_data_sptr(this->add_projdata_sptr);
+    }
+
+    return Succeeded::yes;
+}
+
+Succeeded
+ScatterEstimation::
+set_up_analytic()
+{
+    //TODO : I have most stuff in tmp.
+     error("Analytic recon not implemented yet");
+    return Succeeded::yes;
+}
+
+Succeeded
+ScatterEstimation::
+process_data()
+{
+
+    if (this->set_up() == Succeeded::no)
+    {
+        info("ScatterEstimation: Unsuccessful set up!");
+       return Succeeded::no;
+    }
+
+    const double start_time = this->input_projdata_sptr->get_exam_info_sptr()->get_time_frame_definitions().get_start_time();
+    const double end_time =this->input_projdata_sptr->get_exam_info_sptr()->get_time_frame_definitions().get_end_time();
+
+    float local_min_scale_value = 0.5f;
+    float local_max_scale_value = 0.5f;
+
+    stir::BSpline::BSplineType  spline_type = stir::BSpline::quadratic;
+
+    shared_ptr <ProjData> unscaled_est_projdata_sptr(new ProjDataInMemory(this->scatter_simulation_sptr->get_ExamInfo_sptr(),
+                                                                          this->scatter_simulation_sptr->get_template_proj_data_info_sptr()->create_shared_clone()));
+    scatter_simulation_sptr->set_output_proj_data_sptr(unscaled_est_projdata_sptr);
+
+    shared_ptr<ProjData> scaled_est_projdata_sptr;
+    shared_ptr<ProjData> data_to_fit_projdata_sptr;
+
+    const ChainedBinNormalisation* tmp_chain_multiplicative_binnorm_sptr =
+            dynamic_cast<const ChainedBinNormalisation*>(this->multiplicative_binnorm_sptr.get());
+
+
+    shared_ptr<BinNormalisation> normalisation_factors_sptr;
+    if (!is_null_ptr(tmp_chain_multiplicative_binnorm_sptr ))
+    {
+        normalisation_factors_sptr = tmp_chain_multiplicative_binnorm_sptr->get_first_norm();
+    }
+    else
+    {
+        std::string out_filename = extras_path.get_path() + "ones.hs";
+        shared_ptr<ProjData> normalisation_factors_projdata_sptr(
+                    new ProjDataInterfile(this->input_projdata_sptr->get_exam_info_sptr(),
+                                          this->input_projdata_sptr->get_proj_data_info_sptr()->create_shared_clone(),
+                                          out_filename,
+                                          std::ios::in | std::ios::out | std::ios::trunc));
+        normalisation_factors_projdata_sptr->fill(1.f);
+        normalisation_factors_sptr.reset(new BinNormalisationFromProjData(normalisation_factors_projdata_sptr));
```

Comment:
this would work but seems less wasteful to use `TrivialBinNormalisation` (unless we really need to have a `BinNormalisationFromProjData`)

Reply:
Might work with Trivial ... I will give it a try